src/btree/bt_read.c

```diff
@@ -68,6 +68,129 @@ __row_instantiate(WT_SESSION_IMPL *session,
 	return (0);
 }
 
+/*
+ * __wt_find_lookaside_upd --
+ * 	Scan the lookaside for a record the btree cursor wants to position on.
+ * 	Create an update for the record and return to the caller.
+ */
+int
+__wt_find_lookaside_upd(WT_SESSION_IMPL *session,
+    WT_CURSOR_BTREE *cbt, WT_UPDATE **updp, bool ignore_birthmark)
+{
+	WT_CACHE *cache;
+	WT_CURSOR *cursor;
+	WT_DECL_RET;
+	WT_ITEM *key;
+	WT_ITEM las_key, las_value;
+	WT_REF *ref;
+	WT_UPDATE *upd;
+	wt_timestamp_t durable_timestamp, las_timestamp;
+	size_t incr;
+	uint64_t las_counter, las_pageid, las_txnid;
+	uint32_t las_id, session_flags;
+	uint8_t prepare_state, upd_type;
+	int cmp;
+	bool locked, upd_visible;
+
+	cache = S2C(session)->cache;
+	cursor = NULL;
+	session_flags = 0;		/* [-Werror=maybe-uninitialized] */
+	ref = cbt->ref;
+	las_pageid = ref->page_las->las_pageid;
+	key = &cbt->iface.key;
+	*updp = upd = NULL;
+
+	/* Open a lookaside table cursor. */
+	__wt_las_cursor(session, &cursor, &session_flags);
+
+	/*
+	 * The lookaside records are in key and update order, that is, there
+	 * will be a set of in-order updates for a key, then another set of
+	 * in-order updates for a subsequent key. We scan through all of the
+	 * updates till we locate the one we want.
+	 */
+	WT_PUBLISH(cache->las_reader, true);
+	__wt_readlock(session, &cache->las_sweepwalk_lock);
+	WT_PUBLISH(cache->las_reader, false);
+	locked = true;
+
+	for (ret = __wt_las_cursor_position(cursor, las_pageid);
+	    ret == 0;
+	    ret = cursor->next(cursor)) {
```

Comment:
Don't you need something at the beginning like:

```
WT_ERR(ret);
```

If your `next` call or the initial `__wt_las_cursor_position` fails, don't you want to get out of here?

Reply:
+1, `__wt_find_lookaside_upd()` can fail for a variety of reasons, as can `cursor->next` and we're not handling those errors.

Maybe `WT_ERR_NOTFOUND_OK(ret);` at the end of the loop? That has a problem with the places where we do
```
ret = WT_NOTFOUND;
break;
```
but we don't need to set `ret` at all there, do we? Won't
```
/* Couldn't find a record */
if (upd == NULL && ret == 0)
        ret = WT_NOTFOUND;
```
do what we need?
```

Reply:
good catch. Fixed